src/modules/launcher/Wox.Infrastructure/Image/ImageCache.cs

```diff
@@ -22,7 +23,30 @@ public class ImageCache
                 var i = _data[path];
                 return i;
             }
-            set { _data[path] = value; }
+            set 
+            {
+                _data[path] = value;
+
+                // To prevent the dictionary from drastically increasing in size by caching images, the dictionary size is not allowed to grow more than the permissibleFactor * maxCached size
+                // This is done so that we don't constantly perform this resizing operation and also maintain the image cache size at the same time
+                if (_data.Count > permissibleFactor * MaxCached)
```

Comment:
Can you clarify the difference between the cache and the dictionary? Is cache referring to what is stored in a file versus dictionary referring to what is stored in memory?

Reply:
When launcher is running, the information about images is stored in the dictionary with the path as the key and the Bitmap Imagesource as the value. So the dictionary is in memory (heap) and it continues to grow whenever the user searches for apps/files with unique icons (a hash was calculated using each image to prevent storing duplicates). When the app is closed, the information present in the dictionary would be stored in an Image cache file (on the disk) up to 5000 images which would again be loaded into the dicitonary (memory) at startup. As image Sources are being stored in the dictionary, it is memory intensive compared to storing only integers or strings. Hope this clarifies the difference.